odoo/addons/base/tests/test_expression.py

```diff
@@ -800,6 +799,37 @@ def test_like_wildcards(self):
         countries = self._search(Country, [('name', '=ilike', 'z%')])
         self.assertTrue(len(countries) == 2, "Must match only countries with names starting with Z (currently 2)")
 
+    def test_like_cast(self):
+        Model = self.env['res.partner.category']
+        record = Model.create({'name': 'X', 'color': 42})
+
+        self.assertIn(record, Model.search([('name', '=', 'X'), ('color', 'like', '4')]))
+        self.assertIn(record, Model.search([('name', '=', 'X'), ('color', 'ilike', '4')]))
+        self.assertIn(record, Model.search([('name', '=', 'X'), ('color', 'not like', '3')]))
+        self.assertIn(record, Model.search([('name', '=', 'X'), ('color', 'not ilike', '3')]))
+
+        self.assertNotIn(record, Model.search([('name', '=', 'X'), ('color', 'like', '3')]))
+        self.assertNotIn(record, Model.search([('name', '=', 'X'), ('color', 'ilike', '3')]))
+        self.assertNotIn(record, Model.search([('name', '=', 'X'), ('color', 'not like', '4')]))
+        self.assertNotIn(record, Model.search([('name', '=', 'X'), ('color', 'not ilike', '4')]))
+
+        # like, ilike, not like, not ilike convert their rhs to str
+        self.assertIn(record, Model.search([('name', '=', 'X'), ('color', 'like', '4')]))
```

Comment:
Exactly same assert than a few lines above, without any changes in between? :thinking: 
The comment doesn't help me understand it either

Reply:
Oops, a copy-paste that went wrong.  Thanks for pointing that out!